File: shift_cipher.rb
#!/usr/bin/ruby
# shift_cipher_v0.0.4.rb
# nelbren@nelbren.com
# UseBracketSyntaxIn7lines
def shift_str(x,d)
  r=''; x.each_byte{|c| r += (c == 32 ? c : (c + d > 90 ? c + d - 26 : c + d)).chr }; r
end
def cipher2text(x)
  (0..25).each{|d| p "intento ##{sprintf '%02d', d}: #{shift_str(x,d)}" }
end
cipher2text('PHHW PH DIWHU WKH WRJD SDUWB')

File: shift_cipher.php
#!/usr/bin/php
<?php
# shift_cipher_v0.0.1.php
# nelbren@nelbren.com
# UsePhpIn13lines
function shift_str($x,$d) {
  $r=''; $chars = str_split($x);
  foreach($chars as $c) {
    $o = ord($c);
    $r .= chr($o == 32 ? $o : ($o + $d > 90 ? $o + $d - 26 : $o + $d));
  }
  return $r;
}
function cipher2text($x) {
  for ($d = 0; $d <= 25; $d++) echo "intento " . sprintf('%03d', $d) . ": " . shift_str($x,$d) . "<br>\n";
}
cipher2text('PHHW PH DIWHU WKH WRJD SDUWB');
?>
